### PR TITLE
Support configuring the IMS authority

### PR DESCRIPTION
**Pull request by calebmshafer**
With the introduction of the `itwinjs` scope, we need to provide a way for everyone to change the authority used in the ImsAuthorizationClient. This is a quick way to add in the support as an opt-in but will be further refined during 3.0.

In 3.0, the default of an Bentley IMS based client will be to use the new authority URL. However, this PR will suffice for now on master and be backported directly to 2.19.x.

@skirby1996 is working on setting up integration tests with the new scope using a new set of clients registered directly on the iTwin Platform. Should hopefully have that up soon.

> This is a fork of @laurynasr 's initial PR a few weeks ago.

### CODE REVIEW

File: clients/itwin/src/ImsAuthorizationClient.ts [hunk 1]
```diff
@@ -7,13 +7,15 @@
...
 import { Client } from "./Client";
+import { Config } from "@bentley/bentleyjs-core";
 
 /** @beta */
 export class ImsAuthorizationClient extends Client {
   public static readonly searchKey: string = "IMSOpenID";
 
   public constructor() {
     super();
+    this._url = Config.App.query("imjs_itwin_platform_authority");
```
Comment (aruniverse) [inline clients/itwin/src/ImsAuthorizationClient.ts:18]: if the env var is undefined, do you need to explicitly fallback to imsoidc here? or is that done elsewhere?

**Review by aruniverse — APPROVED** [review]

**Comment by calebmshafer** [comment]
@Mergifyio update

**Comment by calebmshafer** [comment]
@Mergifyio backport release/2.19.x

**Review by ramanujam-raman — APPROVED** [review]

**Comment by calebmshafer** [comment]
@Mergifyio update

**Pull request merged**